common/src/main/java/com/graphicsfuzz/common/util/GloballyTruncateLoops.java

```diff
@@ -39,9 +64,90 @@
    */
   public static void truncate(ShaderJob shaderJob, int loopLimit, String loopCountName,
                               String loopBoundName) {
-    // TODO(https://github.com/google/graphicsfuzz/issues/866): Implement loop limiting
-    //  logic.
-    throw new RuntimeException("Not implemented yet.");
-  }
+    for (TranslationUnit tu : shaderJob.getShaders()) {
+      Declaration firstNonPrecisionDeclaration = null;
+      for (Declaration decl : tu.getTopLevelDeclarations()) {
+        if (decl instanceof PrecisionDeclaration) {
+          continue;
+        }
+        firstNonPrecisionDeclaration = decl;
+        break;
+      }
+      // Add loop bound variable
+      tu.addDeclarationBefore(new VariablesDeclaration(new QualifiedType(BasicType.INT,
+              Arrays.asList(TypeQualifier.CONST)),new VariableDeclInfo(loopBoundName,null,
+              new Initializer(new IntConstantExpr(new Integer(loopLimit).toString())))),
+          firstNonPrecisionDeclaration);
```

Comment:
Google style for Java requires whitespace after commas; can you add please?  (I would have thought CheckStyle would flag this up but apparently it does not.)

Reply:
(And please do this throughout.)

Reply:
Went through the whole code and found only the one instance. And yes, I would have expected CheckStyle to have caught that.